src/Bicep.Core.UnitTests/Diagnostics/LinterRuleTests/MaxNumberAssertsRuleTests.cs

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Bicep.Core.Analyzers.Linter.Rules;
+using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System.Linq;
+
+namespace Bicep.Core.UnitTests.Diagnostics.LinterRuleTests
+{
+    [TestClass]
+    public class MaxNumberAssertsRuleTests : LinterRuleTestsBase
+    {
+        [TestMethod]
+        public void ParameterNameInFormattedMessage()
+        {
+            var ruleToTest = new MaxNumberAssertsRule();
+            ruleToTest.GetMessage(1).Should().Be("Too many asserts. Number of asserts is limited to 1.");
+        }
+
+        private void CompileAndTest(string text, params string[] unusedParams)
+        {
+            AssertLinterRuleDiagnostics(MaxNumberAssertsRule.Code, text, diags =>
+            {
+                if (unusedParams.Any())
+                {
+                    diags.Should().HaveCount(unusedParams.Count());
+
+                    var rule = new MaxNumberAssertsRule();
+                    string[] expectedMessages = unusedParams.Select(p => rule.GetMessage(MaxNumberAssertsRule.MaxNumber)).ToArray();
+                    diags.Select(e => e.Message).Should().Contain(expectedMessages);
+                }
+                else
+                {
+                    diags.Should().BeEmpty();
+                }
+            },
+            new Options(OnCompileErrors.Ignore));
+        }
+
+        [DataRow(@"
+            assert a1 = true
```

Comment:
Could you walk the team through this syntax at an upcoming Bicep Discussions meeting?

Reply:
yes! it's a proposed syntax and could be changed later